src/datasets/demo_only.py

```diff
@@ -1,55 +1,79 @@
 """Create offline datasets by collecting demonstrations."""
 
+import functools
 import logging
-from typing import List, Set
+from typing import Callable, List, Set
+
+import matplotlib
+import matplotlib.pyplot as plt
 
 from predicators.src import utils
 from predicators.src.approaches import ApproachFailure, ApproachTimeout
 from predicators.src.approaches.oracle_approach import OracleApproach
 from predicators.src.envs import BaseEnv
 from predicators.src.settings import CFG
-from predicators.src.structs import Dataset, LowLevelTrajectory, \
-    ParameterizedOption, Task
+from predicators.src.structs import Action, Dataset, LowLevelTrajectory, \
+    ParameterizedOption, State, Task
 
 
 def create_demo_data(env: BaseEnv, train_tasks: List[Task],
                      known_options: Set[ParameterizedOption]) -> Dataset:
     """Create offline datasets by collecting demos."""
-    oracle_approach = OracleApproach(
-        env.predicates,
-        env.options,
-        env.types,
-        env.action_space,
-        train_tasks,
-        task_planning_heuristic=CFG.offline_data_task_planning_heuristic,
-        max_skeletons_optimized=CFG.offline_data_max_skeletons_optimized)
+    assert CFG.demonstrator in ("oracle", "human")
+    if CFG.demonstrator == "oracle":
+        oracle_approach = OracleApproach(
+            env.predicates,
+            env.options,
+            env.types,
+            env.action_space,
+            train_tasks,
+            task_planning_heuristic=CFG.offline_data_task_planning_heuristic,
+            max_skeletons_optimized=CFG.offline_data_max_skeletons_optimized)
+    else:  # pragma: no cover
+        # Disable all built-in keyboard shortcuts.
```

Comment:
you verified that everything still works after you moved this right? sometimes matplotlib is particular about the order in which you change settings vs importing stuff

Reply:
yup